src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs

```diff
@@ -68,11 +68,40 @@ public static bool TryGetEnvironmentVariables(this IResource resource, [NotNullW
     /// <summary>
     ///  Get the environment variables from the given resource.
     /// </summary>
+    /// <remarks>
+    /// This method is useful when you want to make sure the environment variables are added properly to resources, mostly in test situations.
+    /// This method has asynchronous behavior if <paramref name = "applicationOperation" /> be <see cref="DistributedApplicationOperation.Run"/>
+    /// and environment variables were provided from <see cref="IValueProvider"/> otherwise it will be synchronous.
+    /// </remarks>
     /// <param name="resource">The resource to get the environment variables from.</param>
     /// <param name="applicationOperation">The context in which the AppHost is being executed</param>
     /// <returns>The environment variables retrieved from the resource.</returns>
+    /// <example>
+    /// <code>
+    /// using var builder = TestDistributedApplicationBuilder.Create();
```

Comment:
The `TestDistributedApplicationBuilder` type is in this repo only. For the example here you should use `DistributedApplicationTestingBuilder.CreateAsync()`